### PR TITLE
[ALLUXIO-3103] Selectively setOwner during mkdirs

### PR DESCRIPTION
**Pull request by gpang**
https://alluxio.atlassian.net/browse/ALLUXIO-3103

### CODE REVIEW

**Comment by gpang** [comment]
@calvinjia 

If only one is set, I think the semantics are different per UFS. HDFS sets the owner and group in a single call. LocalUFS checks and sets them independently. Object stores do not even set owner/group. I am staying consistent with other parts of our code (https://github.com/Alluxio/alluxio/blob/master/core/common/src/main/java/alluxio/underfs/AtomicFileOutputStream.java#L89).

I think all the UFSes are correct now, or do not use setOwner (object stores).

**Comment by calvinjia** [comment]
Does that mean for HDFS ufs, we could end up having owner bob group bob and end up with owner alice and an empty group?

**Comment by gpang** [comment]
Yes, I think so. However, looking at the HDFS namenode code, it looks like if the strings are `null`, they will be ignored. Maybe this is what we should do for HDFS UFS.

**Comment by gpang** [comment]
@calvinjia I updated the PR to handle the unset user/group better.

File: underfs/local/src/main/java/alluxio/underfs/local/LocalUnderFileSystem.java [hunk 1]
```diff
@@ -293,7 +293,9 @@ public boolean mkdirs(String path, MkdirsOptions options) throws IOException {
         setMode(file.getPath(), options.getMode().toShort());
         FileUtils.setLocalDirStickyBit(file.getPath());
         try {
-          setOwner(file.getPath(), options.getOwner(), options.getGroup());
+          if (options.getOwner() != null || options.getGroup() != null) {
+            setOwner(file.getPath(), options.getOwner(), options.getGroup());
+          }
```
Comment (calvinjia) [inline underfs/local/src/main/java/alluxio/underfs/local/LocalUnderFileSystem.java]: Is this necessary or just for consistency?

Comment (gpang) [reply underfs/local/src/main/java/alluxio/underfs/local/LocalUnderFileSystem.java]: It is unnecessary. I removed the check.

**Pull request merged**